context_errors.go

```diff
@@ -0,0 +1,28 @@
+package errors
+
+import (
+	"context"
+	"fmt"
+)
+
+// ContextError represents a standard error
+// that can also encapsulate a context.
+type ContextError struct {
+	Err error
+	Ctx context.Context
+}
+
+func WrapWithContext(err error, ctx context.Context) *ContextError {
+	return &ContextError{
+		Err: err,
+		Ctx: ctx,
+	}
+}
+
+func (ce *ContextError) Error() string {
+	return fmt.Sprintf("%s", ce.Err)
+}
+
+func (ce *ContextError) Unwrap() error {
```

Comment:
It would be good to also implement `Cause() error` ([see here](https://pkg.go.dev/github.com/pkg/errors#hdr-Retrieving_the_cause_of_an_error)) for better interoperability with `"github.com/pkg/errors"`.